coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,273 @@
+# Coding Guidelines
+## Table of Contents
+
+- [General](#general)
+- [Comments](#comments)
+- [Strings](#strings)
+- [Variables and Attribute Names](#variables-and-attribute-names)
+- [Object Literals](#object-literals)
+- [Indentation and Line Breaks](#indentation-and-line-breaks)
+- [Functions](#functions)
+- [Control Statements](#control-statements)
+- [JSDoc](#jsdoc)
+- [JSX](#jsx)
+	- [Self Contained Elements](#self-contained-elements)
+	- [Non Self Contained Elements](#non-self-contained-elements)
+
+
+## General
+We are using the eslintrc to lint our code. All basic rules are defined there, some important rules are:
+  - Max line length is 80 characters.
+  - Indentation is 2 spaces.
+  - No multiple empty lines.
+
+We prefer a functional approach and try to use the abstractions over
+normal control statements. Functions and code must be
+self-descriptive, use JSDoc blocks only if necessary.
+
+## Comments
+Rules:
+- Comments must be prefixed with a space sign.
+- The `*` sign is a exception to the rule.
+
+Examples:
+```JS
+// Single line Comment
+
+//* A Exception to the space rule
+
+/**
+* Block Comment
+*/
+```
+
+## Strings
+ - Strings are defined with double quotes.
+ - Use template strings if variables are used.
+
+Example:
+```JS
+var string = `Hello ${world}!`;
+```
+
+## Variables and Attribute Names
+Variable names always use camelCase names, except Constants which use UPPER_CASE.
+One var, let or const per variable and line.
+
+Examples:
+```JS
+var camelCase = "{value}";
+let CamelCase = new Object();
+const SOME_CONSTANT = "constant value";
+```
+
+## Object Literals
+Object literals must have no space after the '{' or before the '}' signs.
+
+Valid Example
+```JS
+var objectLiteral = {"key": "value", "key2": "value2"};
+
+var objectLiteral = {
+  "key": "value",
+  "key2": "value2",
+  "key3": "value3",
+  "key4": "value4"
+};
+```
+
+
+## Indentation and Line Breaks
+Rules:
+- Line breaks are only done if needed.
+- Strings are split at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+  ```JS
+    var longString = "A Very long string which is longer then one line, which is 80 characters " +
```

Comment:
Some minor typos:
`A _v_ery long string _that_ is longer then one line, which is 80 characters`
...`line is indented_._`